### ISSUE TITLE
macos El Capitan 10.11.6: Minecraft has exited with a Non-Zero Error code

### ISSUE THREAD

**Issue by tuwtel** [minecraft-linux/mcpelauncher-manifest#327]
When i try to load the launcher on any version i get this error code:
20:21:07 Info  [Launcher] Version: client 4432ade / manifest 04dd224
20:21:07 Info  [Launcher] CPU: GenuineIntel Intel(R) Core(TM)2 Duo CPU     E8135  @ 2.66GHz
20:21:07 Info  [Launcher] CPU supports SSSE3: YES
20:21:07 Trace [Launcher] Loading hybris libraries
20:21:07 Trace [HybrisUtils] Loaded OS library /Applications/Minecraft Bedrock Launcher.app/Contents/Resources/mcpelauncher/libs/native/libfmod.dylib
20:21:07 Trace [HybrisUtils] Loaded OS library libm.dylib
20:21:07 Error [HybrisUtils] Failed to load OS library libz.so.1
20:21:07 Info  [Launcher] Creating window
20:21:07 Trace [Launcher] Loading gamepad mappings: /Applications/Minecraft Bedrock Launcher.app/Contents/Resources/mcpelauncher/gamecontrollerdb.txt
20:21:07 Trace [Launcher] Loading Minecraft library
dyld: lazy symbol binding failed: Symbol not found: _clock_gettime
  Referenced from: /Applications/Minecraft Bedrock Launcher.app/Contents/MacOS/./mcpelauncher-client
  Expected in: /usr/lib/libSystem.B.dylib

dyld: Symbol not found: _clock_gettime
  Referenced from: /Applications/Minecraft Bedrock Launcher.app/Contents/MacOS/./mcpelauncher-client
  Expected in: /usr/lib/libSystem.B.dylib

Process exited with unexpected exit code: 5

**Comment by ChristopherHX** [comment]
Thanks for the crashreport.
I cannot test it for pre 10.14 myself.
10.11 don't have _clock_gettime,
I think it can be stubbed somehow.

**Comment by ChristopherHX** [comment]
Well using a 10.10 sdk shows the error at compile time.
This needs to get rewritten for < 10.12
```
/Users/runner/runners/2.263.0/work/osx-packaging-scripts/osx-packaging-scripts/source/mcpelauncher/libhybris/src/hooks.c:183:29: error: unknown type name 'clockid_t'; did you mean 'clock_t'?
2020-06-10T10:26:19.0519340Z int darwin_my_clock_gettime(clockid_t clk_id, struct timespec *tp) {
2020-06-10T10:26:19.0519830Z                             ^~~~~~~~~
2020-06-10T10:26:19.0520650Z                             clock_t
2020-06-10T10:26:19.0521550Z /Users/runner/runners/2.263.0/work/osx-packaging-scripts/osx-packaging-scripts/../MacOSX-SDKs/MacOSX10.10.sdk/usr/include/sys/_types/_clock_t.h:30:33: note: 'clock_t' declared here
2020-06-10T10:26:19.0521770Z typedef __darwin_clock_t        clock_t;
2020-06-10T10:26:19.0521930Z                                 ^
2020-06-10T10:26:19.0527810Z /Users/runner/runners/2.263.0/work/osx-packaging-scripts/osx-packaging-scripts/source/mcpelauncher/libhybris/src/hooks.c:185:18: error: use of undeclared identifier 'CLOCK_MONOTONIC'
2020-06-10T10:26:19.0528770Z         clk_id = CLOCK_MONOTONIC;
2020-06-10T10:26:19.0529220Z                  ^
2020-06-10T10:26:19.0530280Z /Users/runner/runners/2.263.0/work/osx-packaging-scripts/osx-packaging-scripts/source/mcpelauncher/libhybris/src/hooks.c:186:12: warning: implicit declaration of function 'clock_gettime' is invalid in C99 [-Wimplicit-function-declaration]
2020-06-10T10:26:19.0531230Z     return clock_gettime(clk_id, tp);
2020-06-10T10:26:19.0531650Z            ^
2020-06-10T10:26:19.0545430Z /Users/runner/runners/2.263.0/work/osx-packaging-scripts/osx-packaging-scripts/source/mcpelauncher/libhybris/src/hooks.c:426:18: error: use of undeclared identifier 'mkostemp'; did you mean 'mkstemp'?
2020-06-10T10:26:19.0554750Z     {"mkostemp", mkostemp},
2020-06-10T10:26:19.0567490Z                  ^~~~~~~~
2020-06-10T10:26:19.0571390Z                  mkstemp
2020-06-10T10:26:19.0572610Z /Users/runner/runners/2.263.0/work/osx-packaging-scripts/osx-packaging-scripts/../MacOSX-SDKs/MacOSX10.10.sdk/usr/include/unistd.h:670:6: note: 'mkstemp' declared here
2020-06-10T10:26:19.0573120Z int      mkstemp(char *);
2020-06-10T10:26:19.0573520Z          ^
2020-06-10T10:26:19.0574520Z /Users/runner/runners/2.263.0/work/osx-packaging-scripts/osx-packaging-scripts/source/mcpelauncher/libhybris/src/hooks.c:693:22: error: use of undeclared identifier 'clock_getres'
2020-06-10T10:26:19.0575020Z     {"clock_getres", clock_getres},
2020-06-10T10:26:19.0575420Z                      ^
2020-06-10T10:26:19.0576440Z /Users/runner/runners/2.263.0/work/osx-packaging-scripts/osx-packaging-scripts/source/mcpelauncher/libhybris/src/hooks.c:699:23: error: use of undeclared identifier 'clock_settime'
2020-06-10T10:26:19.0576920Z     {"clock_settime", clock_settime},
2020-06-10T10:26:19.0577310Z                       ^
```

**Comment by ChristopherHX** [comment]
After more bug reports on 10.10, 10.11 support, this bug is finally fixed in Version 0.0.159
I hope it won't come again soon, it's hard to support old macOS versions
